src/modules/inventory/src/inventoryImp.cpp

```diff
@@ -1147,13 +1147,24 @@ void Inventory::SetJsonFieldArray(nlohmann::json& target,
     if (createFields || source.contains(sourceKey))
     {
         const nlohmann::json::json_pointer destPointer(destPath);
-        target[destPointer.parent_pointer()][destPointer.back()] = nlohmann::json::array();
-        nlohmann::json& destArray = target[destPointer];
 
-        if (source.contains(sourceKey) && !source[sourceKey].empty() && !source[sourceKey].is_null() &&
-            source[sourceKey] != EMPTY_VALUE)
+        if (source.contains(sourceKey) && source[sourceKey] != EMPTY_VALUE)
         {
-            destArray.push_back(source[sourceKey]);
+            const auto& value = source[sourceKey];
+
+            if (value.is_null())
+            {
+                target[destPointer] = nullptr;
+            }
+            else
+            {
+                target[destPointer] = nlohmann::json::array();
+                target[destPointer].push_back(value);
+            }
+        }
+        else
+        {
+            target[destPointer] = nullptr;
```

Comment:
is there a way of simplifying this by unifying the target[destPointer] = nullptr ?

Something like this (didn't check it!):
```c++
        target[destPointer] = nullptr;
        if (source.contains(sourceKey) && !source[sourceKey].is_null() && source[sourceKey] != EMPTY_VALUE)
        {
            target[destPointer] = nlohmann::json::array();
            target[destPointer].push_back(value);
        }
```